arithmetization/src/main/java/net/consensys/linea/blockcapture/BlockCapturer.java

```diff
@@ -68,22 +68,27 @@ public void traceStartBlock(BlockHeader blockHeader, BlockBody blockBody) {
     this.reaper.enterBlock(blockHeader, blockBody);
   }
 
+  @Override
+  public void tracePrepareTransaction(WorldView worldView, Transaction transaction) {
+    this.reaper.prepareTransaction(transaction);
+  }
+
   @Override
   public void traceStartTransaction(WorldView worldView, Transaction transaction) {
```

Comment:
we should plug only on `tracePrepareTransaction` besu hook. `traceStartTransaction` is just after, when the upfront gascost (tx + dtat + access list + deployment ) is paid by the sender.

Reply:
We need to use `tracePrepareTransaction()` here.  For some transactions, `traceStartTransaction()` is never called and execution goes straight to `traceEndTransaction()`.  I'm not sure exactly the circumstances, but its either for an invalid transaction or perhaps one that has no code associated with it.  Eitherway, without this, we will get a `NullPointerException` in `traceEndTransaction()` as `txStorage` and `txAddresses` will be `null`.

Reply:
I only found this out from testing, tbh.

Reply:
Anyway, sorry, I guess the point is: I stuck with `traceStartTransation()` because that is what the `BlockCapturer` was originally doing.  However, yes, am happy to just merge into `tracePrepareTransaction()`.

Reply:
yes, we should be plugged only at `tracePrepareTransaction()` and remove `traceStartTransaction()`

Reply:
Yeah sorry for some reason I read you original message the other way around.  Have made the change and just waiting for tests to go green.